src/dev/flang/ast/Match.java

```diff
@@ -422,48 +422,59 @@ else if (c instanceof Block b && b.resultExpression() instanceof Match m)
    *
    * @param fromContract is this an if generated from a contract? (for error messages)
    */
-  public static Match createIf(SourcePosition pos, Expr c, Expr b, Expr elseB, boolean fromContract)
+  public static Expr createIf(SourcePosition pos, Expr c, Expr b, Expr elseB, boolean fromContract)
   {
     if (PRECONDITIONS) require
       (c != null,
        b != null);
 
-    /**
-     * If there is no else / elseif, create a default else
-     * branch returning unit.
-     */
-    if (elseB == null)
+    Expr result;
+    if (c instanceof BoolConst bc)
       {
-        var unit = new Call(pos, FuzionConstants.UNIT_NAME);
-        elseB = new Block(new List<>(unit));
+        result = bc.getCompileTimeConstBool()
+          ? b
+          : elseB;
```

Comment:
I think `elseB` might be `null` here, which then might end up being returned as a result. Or am I missing something?

Reply:
Ugh, I missed that...